cascaritaco/cascarita
# Feature/add user registration and log in endpoints

**Summary**

1. Created User, Role and Group Models. Included migration create the tables in the database. May notice that the User migration contains explicit associations between the Group and Role tables as `groupId` and `roleId` are foreign keys in the user table 
2. Created controllers for each model, with simple create functionality 
3. User model contains functionality to hash and salt user's passwords when creating a new password using the `Bcrypt` package. Also uses it to compare passwords when user logs in. This hashing and compare passwords logic is in the user model file but its used in the `config/passport.js` file. 
4. Cleaned up `app.js` by using router, thus created `router/modelName.routes.js` file system. 


**Testing :**
To create a new user, role , and or  group using postman you would do the following: 

1. if in main, run `git pull`
2. `git checkout  feature/add-user-registration`
3. run migration so you get the new User, Group and Role tables !!!!!: `npx sequelize-cli db:migrate`
4. Double check your database and make sure new migrations ran successfully and you have the new tables 
5. Go to post man and  set to send a `post` request
6. `http://localhost:80/[model-name]/[end-point-that-can-be-found-in-the-route-files-in-the-route-folder` 
7. go to the **Body** tab 
8. select `x-www-form-urlencoded`
9.  set up key value pairs. Reference the model to get the names of the keys. Note if you want to create a new user like in the screenshot provided there needs to be an existing group and role because you need to include a `groupId` and `roleId` 
<img width="1378" alt="Screenshot 2023-12-13 at 3 45 49 PM" src="https://github.com/cascaritaco/cascarita/assets/43281985/ff16e9b1-3d23-4197-8169-3004114aca04">
